stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/ChatFirebaseMessagingService.kt

```diff
@@ -1,65 +1,29 @@
 package io.getstream.chat.android.client.notifications
 
-import android.app.NotificationChannel
-import android.app.NotificationManager
-import android.os.Build
-import androidx.core.app.NotificationCompat
 import com.google.firebase.messaging.FirebaseMessagingService
 import com.google.firebase.messaging.RemoteMessage
 import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.logger.ChatLogger
-import io.getstream.chat.android.client.notifications.handler.NotificationConfig
-import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
 
 internal class ChatFirebaseMessagingService : FirebaseMessagingService() {
     private val logger = ChatLogger.get("ChatFirebaseMessagingService")
-    private val defaultNotificationConfig = NotificationConfig()
 
     override fun onMessageReceived(remoteMessage: RemoteMessage) {
         logger.logD("onMessageReceived(): $remoteMessage")
-        if (!ChatClient.isValidRemoteMessage(remoteMessage, defaultNotificationConfig)) {
-            return
-        }
-        createSyncNotificationChannel()
-        showForegroundNotification(defaultNotificationConfig.smallIcon)
-        GlobalScope.launch(DispatcherProvider.IO) {
-            if (ChatClient.isInitialized) {
-                ChatClient.instance().onMessageReceived(remoteMessage)
-            }
-            stopForeground(true)
+        try {
+            ChatClient.handleRemoteMessage(remoteMessage)
```

Comment:
Service is no longer a foreground service because it doesn't have to load all the data if the user decides to handle push notification themself using `ChatNotificationHandler::onFirebaseMessage`

Reply:
Noo, `isForeground()` checks if `currentState.isAtLeast(Lifecycle.State.STARTED)` and we don't want to shown notifications then

Reply:
Oh sorry. I thought it was meant for the service (a service is in foreground).